src/lib/ext-activitylog.js

```diff
@@ -137,31 +154,58 @@ class Controller {
   }
 
   async init() {
-    this.view.clearLogBtn.addEventListener('clearlog', this);
-    this.view.saveLogBtn.addEventListener('savelog', this);
     this.view.keywordFilter.addEventListener('filterchange', this);
     this.view.extFilter.addEventListener('filterchange', this);
     this.view.viewTypeFilter.addEventListener('filterchange', this);
     this.view.apiTypeFilter.addEventListener('filterchange', this);
+    const searchParams = window.location.search;
 
-    browser.runtime.onMessage.addListener((message) => {
-      const { requestTo, requestType } = message;
+    if (searchParams.includes('page=loadLog&file=')) {
+      this.view.menuContainer.hidden = true;
 
-      if (requestTo !== 'activity-log') {
+      const fileName = searchParams.substring(
+        searchParams.indexOf('file=') + 5
+      );
+      if (!fileName) {
         return;
       }
```

Comment:
We could use URLSearchParams to more easily parse the URL search params, see https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get.

Also, it may be better to make the condition we check in the `if` at line 163 to do not depend from the order of the search params (I mean that we may not need the `page=loadLog` search param and the the `searchParams.includes(...)` condition could be replaced by checking that we do have a `file` search param).

Reply:
Thanks, it's done :-)